components/Article/ArticleGallery.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Gallery } from '@project-r/styleguide/lib/components/Gallery'
+import get from 'lodash.get'
+import memoize from 'lodash.memoize'
+import {
+  imageSizeInfo
+} from 'mdast-react-render/lib/utils'
+
+const shouldInclude = (el) =>
+  get(el, 'identifier', '') === 'FIGURE' && get(el, 'data.excludeFromGallery', false) !== true
+
+const findFigures = (node, acc = []) => {
+  if (get(node, 'children.length', 0) > 0) {
+    node.children.forEach(
+      c => {
+        if (shouldInclude(c)) {
+          acc.push(c)
+        } else {
+          findFigures(c, acc)
+        }
+      }
+    )
+  }
+  return acc
+}
+const getImageProps = (node) => {
+  const src = get(node, 'children[0].children[0].url', '')
+  const alt = get(node, 'children[0].children[0].alt', '')
+  const caption = get(node, 'children[1].children[0].value', '')
+  const credit = get(node, 'children[1].children[1].children[0].value', '')
+  return {
+    src,
+    alt,
+    caption,
+    credit
+  }
+}
+
+class ArticleGallery extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      show: false,
+      startItemSrc: null
+    }
+
+    this.getGalleryItems = memoize(() => {
+      const { article } = this.props
+      return findFigures(article.content)
+        .map(getImageProps)
+        .filter(i => imageSizeInfo(i.src).width > 600)
+    })
+
+    this.renderGallery = () => {
```

Comment:
This render code could live directly inside `render()`, especially now that it's a separate and small component.